src/Workspaces/Remote/ServiceHub/CodeLensOOP/ReferenceCodeLensProvider.cs

```diff
@@ -0,0 +1,228 @@
+                    new object[] { _projectIdGuid, Descriptor.FilePath, Descriptor.ApplicableToSpan.Value.ToTextSpan() }, cancellationToken).ConfigureAwait(false);
+
+                var details = new CodeLensDetailsDescriptor
+                {
+                    Headers = new List<CodeLensDetailHeaderDescriptor>()
+                    {
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.FilePath },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.LineNumber },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ColumnNumber },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ReferenceText },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ReferenceStart },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ReferenceEnd },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ReferenceLongDescription },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.ReferenceImageId },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.TextBeforeReference2 },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.TextBeforeReference1 },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.TextAfterReference1 },
+                        new CodeLensDetailHeaderDescriptor() { UniqueName = ReferenceEntryFieldNames.TextAfterReference2 },
+                    },
+                    Entries = referenceLocationDescriptors.SelectAsArray(referenceLocationDescriptor =>
+                    {
+                        ImageId imageId = default;
+                        if (referenceLocationDescriptor.Glyph.HasValue)
+                        {
+                            var moniker = Util.GetImageMoniker(referenceLocationDescriptor.Glyph.Value);
+                            imageId = new ImageId(moniker.Guid, moniker.Id);
+                        }
+
+                        return new CodeLensDetailEntryDescriptor()
+                        {
+                            // use default since reference codelens don't require special behaviors
+                            NavigationCommand = null,
+                            NavigationCommandArgs = null,
+                            Tooltip = null,
+                            Fields = new List<CodeLensDetailEntryField>()
+                            {
+                                // file path
+                                new CodeLensDetailEntryField() { Text =   Descriptor.FilePath },
```

Comment:
strange spacing.

Reply:
multiline collection initializer will be left as user wrote. so if you have formatting suggestion, write down. I will use that.

Reply:
My suggestion woudl be to remove the extra spaces after hte ```=```.  None of the other lines have it :)